consensus/ibft/src/integration-test/java/tech/pegasys/pantheon/consensus/ibft/tests/RoundChangeTest.java

```diff
@@ -0,0 +1,319 @@
+    final SignedData<RoundChangePayload> rc2 = transmitter.injectRoundChange(targetRound, empty());
+    final SignedData<RoundChangePayload> rc3 = transmitter.injectRoundChange(targetRound, empty());
+
+    assertPeersReceivedNoMessages(roles.getAllPeers());
+  }
+
+  @Test
+  public void subsequentRoundChangeMessagesFromPeerDoNotOverwritePriorMessage() {
+    final long ARBITRARY_BLOCKTIME = 1500;
+
+    final ConsensusRoundIdentifier bestPreparedRound = new ConsensusRoundIdentifier(1, 2);
+    final ConsensusRoundIdentifier targetRound = new ConsensusRoundIdentifier(1, 3);
+    final Block bestPreparedBlock = context.createBlockForProposal(2, ARBITRARY_BLOCKTIME);
+
+    // Create a roundChange containing a PreparedCertificate
+    final SignedData<RoundChangePayload> rc1 =
+        roles
+            .getProposer()
+            .injectRoundChange(
+                targetRound,
+                Optional.of(
+                    new PreparedCertificate(
+                        roles
+                            .getNonProposingPeers()
+                            .get(1)
+                            .getMessageFactory()
+                            .createSignedProposalPayload(bestPreparedRound, bestPreparedBlock),
+                        Lists.newArrayList(
+                            roles
+                                .getProposer()
+                                .getMessageFactory()
+                                .createSignedPreparePayload(
+                                    bestPreparedRound, bestPreparedBlock.getHash()),
+                            roles
+                                .getNonProposingPeers()
+                                .get(0)
+                                .getMessageFactory()
+                                .createSignedPreparePayload(
+                                    bestPreparedRound, bestPreparedBlock.getHash())))));
+
+    // Attempt to override the previously received RoundChange (but now without a payload).
+    roles.getProposer().injectRoundChange(targetRound, empty());
+
+    final SignedData<RoundChangePayload> rc2 =
+        roles.getNonProposingPeer(0).injectRoundChange(targetRound, empty());
+    final SignedData<RoundChangePayload> rc3 =
+        roles.getNonProposingPeer(1).injectRoundChange(targetRound, empty());
+
+    final Block expectedBlockToPropose =
+        context.createBlockForProposal(targetRound.getRoundNumber(), ARBITRARY_BLOCKTIME);
+
+    final SignedData<NewRoundPayload> expectedNewRound =
+        localNodeMessageFactory.createSignedNewRoundPayload(
+            targetRound,
+            new RoundChangeCertificate(Lists.newArrayList(rc1, rc2, rc3)),
+            localNodeMessageFactory.createSignedProposalPayload(
+                targetRound, expectedBlockToPropose));
+
+    assertPeersReceivedExactly(roles.getAllPeers(), expectedNewRound);
+  }
+
+  @Test
+  public void messagesFromPreviousRoundAreDiscardedOnTransitionToFutureRound() {}
```

Comment:
looks like this test hasn't been implemented

Reply:
done.